From da23693aee5b8226f0c475988ed88c11994a1364 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <philippe.gerum@exail.com>
Date: Wed, 25 Jun 2025 19:28:34 +0200
Subject: [PATCH 971/985] evl/net: packet: use bound device directly in
 transmit

We don't need to fetch the output device via the get_netif()
indirection from evl_net_proto, since by definition, we must be
calling back into our own implementation of this handler.

Call the latter directly, instead of using the indirection.

Signed-off-by: Philippe Gerum <philippe.gerum@exail.com>
---
 kernel/evl/net/packet/packet.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/kernel/evl/net/packet/packet.c b/kernel/evl/net/packet/packet.c
index e0842b841..f42585069 100644
--- a/kernel/evl/net/packet/packet.c
+++ b/kernel/evl/net/packet/packet.c
@@ -384,7 +384,7 @@ static struct net_device *find_xmit_device(struct evl_socket *esk,
 			if (namelen)
 				return ERR_PTR(-EINVAL);
 
-			dev = esk->proto->get_netif(esk);
+			dev = get_netif_packet(esk);
 		} else {
 			if (namelen < sizeof(addr))
 				return ERR_PTR(-EINVAL);
@@ -401,7 +401,7 @@ static struct net_device *find_xmit_device(struct evl_socket *esk,
 			dev = evl_net_get_dev_by_index(esk->net, addr.sll_ifindex);
 		}
 	} else {
-		dev = esk->proto->get_netif(esk);
+		dev = get_netif_packet(esk);
 	}
 
 	if (dev == NULL)
@@ -667,7 +667,7 @@ static __poll_t poll_packet(struct evl_socket *esk,
 	if (!list_empty(&esk->input) || evl_test_socket_iots(esk))
 		ret = POLLIN|POLLRDNORM;
 
-	dev = esk->proto->get_netif(esk);
+	dev = get_netif_packet(esk);
 	if (dev) {
 		est = dev->oob_state.estate;
 		evl_poll_watch(&est->poll_head, wait, NULL);
-- 
2.30.2

